### PR TITLE
Added option for serialization of grids.

### PR DESCRIPTION
**Pull request by luca-heltai**
This PR allows one to write and read the `ar` format, which stands for boost::archive. This is what is generated by tria.save, and tria.load. 

Using this format it is possible to save and restore locally refined grids.

### CODE REVIEW

**Comment by luca-heltai** [comment]
(feature requested by Daniele Boffi)

File: source/parsed_grid_generator.cc [hunk 1]
```diff
@@ -19,6 +19,8 @@
 #include <deal.II/grid/grid_generator.h>
 #include <deal.II/grid/grid_in.h>
 #include <deal.II/grid/grid_out.h>
+#include <boost/archive/text_oarchive.hpp>
```
Comment (asartori86) [inline source/parsed_grid_generator.cc:22]: dobbiamo aggiungere un guard?

Comment (luca-heltai) [inline source/parsed_grid_generator.cc:22]: No. E' incluso in deal.II.

Comment (asartori86) [inline source/parsed_grid_generator.cc:22]: nemmeno DEAL_II_WITH_BOOST?

**Pull request merged**